doc/modules/neural_networks_supervised.rst

```diff
@@ -0,0 +1,357 @@
+
+Given a set of training examples :math:`(x_1, y_1), (x_2, y_2), \ldots, (x_n, y_n)` 
+where :math:`x_i \in \mathbf{R}^n` and :math:`y_i \in \{0, 1\}`, a one hidden 
+layer mlp learns the score function :math:`f(x) = W_2^T g(W_1^T x + b_1) + b_2` 
+where :math:`W_1, W_2 \in \mathbf{R}^m` and :math:`b_1, b_2 \in \mathbf{R}` are 
+model parameters. :math:`W_1, W_2` represent the weights of the input layer and 
+hidden layer, resepctively; and :math:`b_1, b_2` represent the bias vectors added 
+to the hidden layer and the output layer, respectively. 
+:math:`g(\cdot) : R \rightarrow R` is the activation function, set by default as 
+the hyperbolic tan. It is given as,
+
+.. math::
+      g(x)= \frac{e^x-e^{-x}}{e^x+e^{-x}}
+
+For binary classification, :math:`f(x)` passes through the logistic function
+:math:`g(x)= 1/(1+e^{-x})` to get  output values between zero and one. A threshold,
+set to 0.5, would assign samples of outputs larger or equal 0.5 to the positive 
+class, and the rest to the negative class.
+
+If there are more than two classes, :math:`f(x)` would instead pass through
+the softmax function, which is written as,
+
+.. math::
+      \text{Softmax} = \frac{\exp(W_i^Tx)}{\sum_{l=1}^k\exp(W_l^Tx)} 
+
+where :math:`W_i` contains the weights incident to the output neuron representing
+class :math:`i`, and :math:`K` is the number of classes. The result is a vector 
+containing the probabilities that sample :math:`x` belong to each class. The 
+output is the class with the highest probability.
+
+In regression, the output remains as :math:`f(x)`; therefore, there is no
+output activation function.
+
+MLP uses different loss functions depending on the problem type. The loss 
+function for classification is Cross-Entropy, which in binary case is given as,
+
+.. math::
+
+    Loss(x,y,W) = -y \ln {f(x)}+(1-y)\ln{(1-f(x))} + \alpha ||W||_2^2
+
+where :math:`\alpha ||W||_2^2` is an L2-norm regularization term (aka penalty)
+that penalizes model complexity; and :math:`\alpha > 0` is a non-negative 
+hyperparameter that controls the magnitude of the penalty term.
+
+For regression, MLP uses the Square Error loss function; written as,
+
+.. math::
+
+    Loss(x,y,W) = \frac{1}{2}||f(x) - y ||_2^2 + \alpha ||W||_2^2
+
+
+Starting from initial random weights, multi layer perceptron (MLP) minimizes 
+the loss function by repeatedly updating these weights. After computing
+the loss function value, a backward pass propagates it from the output layer 
+to the initial layer, providing each weight parameter with an update value 
+meant to decrease the loss function.
+
+In gradient descent, the weight gradient :math:`\nabla W_{loss}` with respect
+to the loss function is computed and added to the corresponding :math:`W`.
+More formally, this is expressed as,
+
+.. math::
+
+    W^{i+1} = W^i + \epsilon \nabla W^i_{loss}
+
+
+where :math:`i` is the iteration step, and :math:`\epsilon` is the learning rate 
+with a value ranging from 0 and 1. 
```

Comment:
I have used it with batch normalization. Normal problem is everything blows up to NaN but it can go as high as you want to go depending on the problem